### PR TITLE
Implement proper HttpMiddleware.basicAuth

### LINKED ISSUE #70: Implement proper HttpMiddleware.basicAuth
Currently, there is a "proof of concept" implementation of an HttpMiddleware that does basic authentication.

This implementation is known to be incomplete and has not been tested. The implementation should be completed and have a test suite that ensures both that it functions as specified, and that it is resistant to various attacks, including malformed data.

Possibly, we need a new `Sensitive` data type that is a byte-array CharSequence that can be converted to a string (and created from a string) but whose toString() does not render the information. Then we can use this type instead of `String` for the password.

This ticket will be complete when `basicAuth` is fully functioning implementation of the spec and has a test suite with good test coverage.

### PR DESCRIPTION
**Pull request by jaliss**
This PR fixes #70

### CODE REVIEW

**Comment by jaliss** [comment]
Pending so far: 

 - Implement `Sensitive`
- Spec for the basic auth middleware.

File: core/src/main/scala/zio/web/http/auth/BasicAuth.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  object AuthParams {
+
+    def create(realm: String, header: String): Option[AuthParams] =
+      if (!header.startsWith("Basic")) {
+        None
+      } else {
+        val data = header.drop("Basic ".length).trim
+
+        // TODO: base64 decode data
```
Comment (jaliss) [inline core/src/main/scala/zio/web/http/auth/BasicAuth.scala:26]: this TODO was in the original code, did we forget to remove it or was the intent to decode using something different than `java.util.Base64`?

**Comment by jdegoes** [comment]
@jaliss Is this ready for a final review? Looks like maybe some tests were failing on CI, but would like to get this merged in as soon as possible!

**Comment by jdegoes** [comment]
@jaliss Can we move this to the (newly-formed) `middleware` package?

**Comment by jaliss** [comment]
@jdegoes sure, I'll move this to the new `middleware` package.

**Comment by jaliss** [comment]
Closed in favor of #115

**Pull request closed without merging**